core/modules/user/components/UserAccount.ts

```diff
@@ -150,25 +151,36 @@ export const UserAccount = {
           newPassword: this.password
         })
       }
+      updatedProfile = pick(updatedProfile, config.users.allowModification)
       this.exitSection(null, updatedProfile)
     },
     exitSection (event, updatedProfile) {
       this.$bus.$emit('myAccount-before-updateUser', updatedProfile)
-      if (!updatedProfile) {
-        this.currentUser = Object.assign({}, this.$store.state.user.current)
-        this.userCompany = this.getUserCompany()
-        this.changePassword = false
-        this.oldPassword = ''
-        this.password = ''
-        this.rPassword = ''
-        if (!this.userCompany.company) {
-          this.addCompany = false
+      userHooks.afterUserProfileUpdated(async (event) => {
+        if (event.resultCode === 200) {
+          if (!updatedProfile) {
+            this.currentUser = Object.assign({}, this.$store.state.user.current)
+            this.userCompany = this.getUserCompany()
+            this.changePassword = false
+            this.oldPassword = ''
+            this.password = ''
+            this.rPassword = ''
+            if (!this.userCompany.company) {
+              this.addCompany = false
+            }
+            this.remainInEditMode = false
+          }
+          if (!this.remainInEditMode) {
+            this.isEdited = false
+          }
+        } else {
+          await this.$store.dispatch('notification/spawnNotification', {
```

Comment:
Do we need `await`? I see no Promise in spawnNotification:
```ts
spawnNotification ({ commit, state, dispatch }, notification: NotificationItem): NotificationItem {
      if (state.notifications.length > 0 &&
        state.notifications[state.notifications.length - 1].message === notification.message
      ) {
        return
      }

      const id = Math.floor(Math.random() * 100000)
      const newNotification = { id, ...notification }

      commit('add', newNotification)

      if (!newNotification.hasNoTimeout) {
        setTimeout(() => {
          dispatch('removeNotificationById', id)
        }, newNotification.timeToLive || 5000)
      }

      return newNotification
    }
```

Reply:
nope, removed